### DISCUSSION TITLE
voxel-dependent reference dose in optimization

### DISCUSSION THREAD

**Discussion by ehssal** [e0404/matRad#526] [Q&A]
Hi there,

I am wondering whether in "function obj = matRad_SquaredDeviation(penalty,dRef)", dRef could be defined as a voxel-dependent vector so that we could potentially set a different reference dose for every voxel in the VOI. I tried to pass a cell array as dRef but that caused errors in the optimization routine.

Thanks!

**Comment by wahln** [comment]
Hi,
we actually looked at this just a few days back.
If you provide the "raw" objective with a d_ref vector and prototype an evaluation, it does in principle work. 
But at the moment there's some other minor  things breaking this:
- a built-in check for the reference dose being a scalar in the SquaredDevation objective
- transforming the dose prescriptions for biological / effect-based optimization expects scalar reference doses
- Display in the GUI needs to be adapted
- Also one thing to remember is that one needs to take care that the d_ref vector has the length of the number of voxels in the dose grid for the given structure.

Since we are looking into it anyway I will probably implement a prototype for this to work with code (not GUI) within the next days and push it to the dev branch.

**Reply by ehssal** [reply]
Thank you for the response. This will be a useful feature for adaptive planning. I attempted to make changes in the code every time the getDoseParameters was called so that it accepts a cell array as opposed to a scalar, but it seemed to cause additional errors. Also, I was not sure whether gradient calculation changes are needed as well. I look forward to seeing the prototype. Thanks!
